src/mame/video/fmtowns.cpp

```diff
@@ -1104,7 +1108,11 @@ void towns_state::towns_crtc_draw_scan_layer_hicolour(bitmap_rgb32 &bitmap,const
 				off &= 0x3ffff;  // 2 layers
 			else
 				off &= 0x7ffff;  // 1 layer
-			colour = (m_towns_gfxvram[off+(layer*0x40000)+1] << 8) | m_towns_gfxvram[off+(layer*0x40000)];
+			// TODO: figure out how to wrap when there are sprites on layer 1
+			if ((layer == 0 || !(m_video.towns_sprite_reg[1] & 0x80)) && off >= (line * linesize) && off - (line * linesize) >= linesize)
```

Comment:
Can you turn this check into a lambda or something?  Repeating it is a liability.

Reply:
I have rewritten the checks using lambda expressions. The original code is already quite repetitive (it has one "if" block for each zoom level) but at least this shouldn't add too much fuel to the fire.